tests/Fixer/Operator/StandardizeIncrementFixerTest.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Operator;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\Operator\StandardizeIncrementFixer
+ */
+final class StandardizeIncrementFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixCases()
+    {
+        return [
+            [
+                '<?php ++$i;',
+                '<?php $i = $i + 1;',
+            ],
+            [
+                '<?php ++$i;',
+                '<?php $i=$i+1;',
+            ],
+            [
+                '<?php ++$i;',
+                '<?php $i   =  $i  +   1;',
+            ],
+            [
+                '<?php ++$i;',
+                '<?php $i += 1;',
+            ],
+            [
+                '<?php ++$i;',
+                '<?php $i+=1;',
+            ],
+            [
+                '<?php --$i;',
+                '<?php $i = $i - 1;',
+            ],
+            [
+                '<?php --$i;',
+                '<?php $i=$i-1;',
+            ],
+            [
+                '<?php --$i;',
+                '<?php $i -= 1;',
+            ],
+            [
+                '<?php --$i;',
+                '<?php $i-=1;',
+            ],
+            [
+                '<?php ++$i;',
+                '<?php $i /* foo */ = $i /* bar */ + 1;',
+            ],
+            [
+                '<?php for ($i = 0; $i < $n; ++$i) {};',
+                '<?php for ($i = 0; $i < $n; $i = $i + 1) {};',
+            ],
+            [
+                '<?php for ($i = 0; $i < $n; ++$i) {};',
+                '<?php for ($i = 0; $i < $n; $i += 1) {};',
+            ],
+            [
+                '<?php foo_bar(++$i);',
+                '<?php foo_bar($i = $i + 1);',
+            ],
+            [
+                '<?php $foo->bar = $foo->bar + 1;',
```

Comment:
Any thoughts on this and the few below? If the `__set` or `__get` methods were implemented in a way that it would not set/get the same value then it would make this fixer risky

Reply:
Seems to be fine: https://3v4l.org/HlOvJ.

Reply:
Yep, you're right. Dunno what I was thinking

Reply:
Can anyone give me any pointers on how I would detect if it should be shortened in cases like this?

`$foo->bar = $foo->bar + 1` should be `++$foo->bar`

`$foo->bar[baz()] = $foo->bar[baz()] + 1` should stay the same

`$foo->bar[baz()] += 1` should be `++$foo->bar[baz()]` because `baz` will only be called once in both circumstances

`$foo->bar[$baz->qux] = $foo->bar[$baz->qux] + 1` should stay the same because `qux` may changed

`$foo->bar[$baz->qux] += 1` should be `++$foo->bar[$baz->qux]` because `qux` will only be accessed once in both circumstances